GoogleApiWrappers/src/Google.Apis.Storage.v1.IntegrationTests/UploadObjectTest.cs

```diff
@@ -0,0 +1,122 @@
+
+        [Fact]
+        public void SimpleUpload()
+        {
+            var name = GenerateName();
+            var contentType = "application/octet-stream";
+            var source = GenerateData(100);
+            var result = s_config.Client.UploadObject(s_bucket, name, contentType, source);
+            Assert.Equal(s_bucket, result.Bucket);
+            Assert.Equal(name, result.Name);
+            Assert.Equal(contentType, result.ContentType);
+            ValidateData(source, name);            
+        }
+
+        [Fact]
+        public void UploadWithObject()
+        {
+            var destination = new Object
+            {
+                Bucket = s_bucket,
+                Name = GenerateName(),
+                ContentType = "test/type",
+                ContentDisposition = "attachment",
+                Metadata = new Dictionary<string, string> { { "x", "y" } }
+            };
+            var source = GenerateData(100);
+            var result = s_config.Client.UploadObject(destination, source);
+            Assert.NotSame(destination, result);
+            Assert.Equal(destination.Name, result.Name);
+            Assert.Equal(destination.Bucket, result.Bucket);
+            Assert.Equal(destination.ContentType, result.ContentType);
+            Assert.Equal(destination.ContentDisposition, result.ContentDisposition);
+            Assert.Equal(destination.Metadata, result.Metadata);
+            ValidateData(source, destination.Name);
+        }
+
+        [Fact]
+        public async Task UploadAsyncWithProgress()
+        {
+            var chunks = 2;
+            var name = GenerateName();
+            var contentType = "application/octet-stream";
+            var source = GenerateData(UploadObjectOptions.MinimumChunkSize * chunks);
+            int progressCount = 0;
+            var progress = new Progress<IUploadProgress>(p => progressCount++);
+            var result = await s_config.Client.UploadObjectAsync(s_bucket, name, contentType, source,
+                new UploadObjectOptions { ChunkSize = UploadObjectOptions.MinimumChunkSize },
+                CancellationToken.None, progress);
+            Assert.Equal(chunks + 1, progressCount); // Should start with a 0 progress
+            Assert.Equal(name, result.Name); // Assume the rest of the properties are okay...
+            ValidateData(source, name);
+        }
+
+        [Fact]
+        public void ReplaceObject()
+        {
+            // This test relies on versioning being enabled for bucket {project_name}_integrationtests-2
+            var name = GenerateName();
+            var contentType = "application/octet-stream";
+            var source1 = GenerateData(100);
+            var firstVersion = s_config.Client.UploadObject(s_bucket, name, contentType, source1);
+            ValidateData(source1, name);
+            var source2 = GenerateData(50);
+            firstVersion.ContentType = "application/x-replaced";
```

Comment:
This is a bit confusing, shouldn't you test that `firstVersion.ContentType` matches the specified `contentType`, that is `application/octet-stream` on return?